test/tests.nix

```diff
@@ -0,0 +1,153 @@
+# Integration tests, can be run without internet access.
+
+{ scenario ? "default" }:
+
+import ./lib/make-test.nix scenario (
+{ config, pkgs, lib, ... }: with lib;
+let testEnv = rec {
+  cfg = config.services;
+  mkIfTest = test: mkIf (config.tests.${test} or false);
+
+  baseConfig = {
+    imports = [
+      ./lib/test-lib.nix
+      ../modules/modules.nix
+      ../modules/secrets/generate-secrets.nix
+    ];
+
+    config = {
+      tests.bitcoind = cfg.bitcoind.enable;
+      services.bitcoind = {
+        enable = true;
+        extraConfig = mkIf config.test.noConnections "connect=0";
+      };
+
+      tests.clightning = cfg.clightning.enable;
+
+      tests.spark-wallet = cfg.spark-wallet.enable;
+
+      tests.nanopos = cfg.nanopos.enable;
+
+      tests.lnd = cfg.lnd.enable;
+      services.lnd.listenPort = 9736;
+
+      tests.lightning-loop = cfg.lightning-loop.enable;
+      # Needed because we must control when lightning-loop starts so it doesn't
+      # fail before we run commands in the nb-lightning-loop netns
+      systemd.services.lightning-loop.wantedBy = mkIfTest "lightning-loop" (mkForce []);
+
+      tests.electrs = cfg.electrs.enable;
+
+      tests.liquidd = cfg.liquidd.enable;
+      services.liquidd.extraConfig = mkIf config.test.noConnections "noconnect=1";
+
+      tests.btcpayserver = cfg.btcpayserver.enable;
+      services.btcpayserver.lightningBackend = "lnd";
+      # Needed to test macaroon creation
+      environment.systemPackages = mkIfTest "btcpayserver" (with pkgs; [ openssl xxd ]);
+
+      tests.joinmarket = cfg.joinmarket.enable;
+      services.joinmarket.yieldgenerator = {
+        enable = config.services.joinmarket.enable;
+        customParameters = ''
+          txfee = 200
+          cjfee_a = 300
+        '';
+      };
+
+      tests.backups = cfg.backups.enable;
+
+      # To test that unused secrets are made inaccessible by 'setup-secrets'
+      systemd.services.generate-secrets.postStart = mkIfTest "security" ''
+        install -o nobody -g nogroup -m777 <(:) /secrets/dummy
+      '';
+    };
+  };
+
+  scenarios = {
+    base = baseConfig; # Included in all scenarios
+
+    default = scenarios.secureNode;
+
+    full = {
```

Comment:
How about adding a comment describing what the scenario is supposed to test? Looks like `full` is the full set of nixos modules with the exception of secure-node.nix.

Reply:
Fixed.